docs/data/data-grid/editing-legacy/editing.md

```diff
@@ -0,0 +1,272 @@
+```
+
+{{"demo": "BasicRowEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Start editing
+
+If a cell is editable and has focus, any of the following interactions will start the edit mode of the corresponding row:
+
+- A <kbd class="key">Enter</kbd> keydown
+- A double click on the cell
+- A call to `apiRef.current.setRowMode(id, 'edit')`.
+
+  ```tsx
+  /**
+    * Sets the mode of a row.
+    * @param {GridRowId} id The id of the row.
+    * @param {GridRowMode} mode Can be: `"edit"`, `"view"`.
+    */
+  setRowMode: (id: GridRowId, mode: GridRowMode) => void;
+  ```
+
+### Stop editing
+
+If a row is in edit mode and one of its cells is focused, any of the following interactions will stop the edit mode:
+
+- A <kbd class="key">Escape</kbd> keydown. It will also roll back changes done in the row.
+- A <kbd class="key">Enter</kbd> keydown. It will also save and goes to the cell at the next row at the same column.
+- A mouse click outside the row
+- A call to `apiRef.current.setRowMode(id, 'view')`.
+
+### Controlled editing
+
+The `editRowsModel` prop lets you control the editing state.
+You can handle the `onEditRowsModelChange` callback to control the `GridEditRowsModel` state.
+
+{{"demo": "RowEditControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Conditional validation [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+Having all cells of a row in edit mode allows validating a field based on the value of another one.
+To accomplish that, start by adding the `preProcessEditCellProps` as explained in the [cell editing](#client-side-validation).
+When the callback is called, use the API to check the value of the other field and decide if the current value is valid or not.
+Return a new object contaning `params.props` and the `error` attribute with the validation status.
+Once at the least one field has the `error` attribute equals to true no new value will be committed.
+
+**Note:** For server-side validation, the same [approach](#server-side-validation) from the cell editing can be used. The data grid will wait for all promises to resolve before commiting.
+
+The following demo requires a value for the "Payment method" column if the "Is paid?" column was checked.
+
+{{"demo": "ConditionalValidationGrid.js", "disableAd": true, "bg": "inline", "defaultCodeOpen": false}}
+
+> The conditional validation can also be implemented with the [controlled editing](#controlled-editing-2).
+> This approach can be used in the free version of the DataGrid.
+> The only limitation is that it does not work with the `singleSelect` column type.
+
+### Control with external buttons [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+You can [disable the default behavior](/components/data-grid/events/#disabling-the-default-behavior) of the grid and control the row edit using external buttons.
+
+Here is shown how a full-featured CRUD can be created.
```

Comment:
Updated. I only removed "app" because it's not a real app. It's more of a component that lets create, read, update, etc. records like in a business application.